src/Symfony/Bridge/Twig/Extension/TranslationExtension.php

```diff
@@ -54,6 +56,16 @@ public function getTranslator(): TranslatorInterface
         return $this->translator;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('_t', [$this, 'createTranslatable']),
```

Comment:
im not sure having this makes sense in twig :confused: IMHO it's always `{{ var|trans }}`, where `var` is either a string or Translatable.

Reply:
or put different, why would you create a Translatable in twig, if you can do `{{ 'string'|trans(params, domain)` already

Reply:
I have 2 reasons for this:
1. When using a fallback/default message that might have a different set of params or in a different domain. Consider this:
```twig
{{ message|default(_t(fallback, {'%param%': value}))|trans }}
```
In this scenario `message` is a translatable object/string and we are able to supply a fallback translatable if whatever we're expecting to have isn't set. We shouldn't pass the params to the `trans` filter as they don't make sense in the context of some kind of message, but do in the fallback, or if we wanted to set the domain on the fallback for example, but is different from what we want the default to be. For me, this just removes complexity from what we would have to do otherwise.

2. Personal preference - for myself at least it puts more context between the message we're trying to create and the params/domain associated. This reduces cognitive load having to make a connection between the message and the data the message requires.

I don't think this addition changes how we would intend people to translate, but offers another approach. I'm happy to hear your thoughts on this.

Reply:
hm, fallback scenario looks elegant i agree :+1: